packages/w3up-client/src/client.js

```diff
@@ -311,4 +311,46 @@ export class Client extends Base {
       proofs: options.proofs,
     })
   }
+
+  /**
+   * Removes association of a content CID with the space. Optionally, also removes
+   * association of CAR shards with space.
+   *
+   * @param {import('multiformats').UnknownLink} contentCID
+   * @param {object} [options]
+   * @param {boolean} [options.shards]
+   */
+  async remove(contentCID, options = {}) {
+    // Remove association of content CID with selected space.
+    let upload
+    try {
+      upload = await this.capability.upload.remove(contentCID)
```

Comment:
I would expect this to be run only after we remove all the shards instead of doing it before we remove the shards. That is because is one of the operations fails, caller will still have an opportunity to re-try. On the other hand if we remove the upload and fail to remove one of the shards, there will be no way for the user to retry or clear the shards.

If there is a good reason to do it this way can you please add code comments elaborating it. Otherwise I suggest to try removing all the shards and if they succeed only after remove the root.

Reply:
There is no real reason, I basically copied what was done in CLI given that functionality is there. I will see who did the change and ask for feedback to align on it

Reply:
Ok, so this code was added over a year ago https://github.com/web3-storage/w3cli/pull/20

I guess we would be ok with changing for suggested, but let me ask @olizilla if there is any reason on their head

Reply:
Honestly, I think we could consider not facilitating a high level function to client as it is not an atomic op, may have mutations on partial failure, and as pointed out when part of multiple uploads may also be an issue.

Leaving this for the advanced function prefixed by capabilities could be a way to deal with this, but we would need to explain users upload and store, and update the docs (and of course make CLI no delete like this, but also in a granular way)